File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis.VisualBasic/PseudoVariableExpressions.cs
namespace Microsoft.CodeAnalysis.VisualBasic
{
	internal abstract class PseudoVariableExpressions
	{
		internal abstract BoundExpression GetAddress(BoundPseudoVariable variable);

		internal abstract BoundExpression GetValue(BoundPseudoVariable variable, DiagnosticBag diagnostics);
	}
}

File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis/VisualBasic/PseudoVariableExpressions.vb
Imports Microsoft.CodeAnalysis
Imports System

Namespace Microsoft.CodeAnalysis.VisualBasic
	Friend MustInherit Class PseudoVariableExpressions
		Protected Sub New()
			MyBase.New()
		End Sub

		Friend MustOverride Function GetAddress(ByVal variable As BoundPseudoVariable) As BoundExpression

		Friend MustOverride Function GetValue(ByVal variable As BoundPseudoVariable, ByVal diagnostics As DiagnosticBag) As BoundExpression
	End Class
End Namespace